module-2/backend/week_one.md

```diff
@@ -5,30 +5,114 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+  * Using REST makes the code readable to everyone, even if they're new to the project. It makes the code logical to everyone
+  * familiar with the conventions. *
 6. What types of variables are accessible in our view templates without explicitly passing them?
+  *The view templates can access local and instance variables.*
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
     erb :index
   end
   ```
-
+  ***
+  ```ruby
+    get '/horses' do
+      @count = 1
+      erb :index
+    end
+    ```
+    ***
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+    ```ruby
+    get '/horses' do
+      erb :index, locals: {name: "Mr. Ed"}
+    end
+    ```
 9. What's the purpose of ERB?
+  *To have access to ruby to render dynamic views. *
 10. Why do I need a development AND test database?
+  *The test database should be clean so that tests run without data that may have been entered during development. *
 11. What's responsive design?
+  *It's design that works for both desktop and mobile because the screen size is flexible. It responds to the screen size changing. *
 12. What is CRUD and why is it important?
+  * CRUD (Create, Read, Update, Delete) is the foundation for web development. It provides the basic functionality needed for
+  a web app. *
 13. What does HTTP stand for? 
+  *Hyper Text Transfer Protocol*
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+  * 1: <%= test %> 2. <% test %> *
+  * Use 1 when you want to render the Ruby. Use 2 when you want to evaluate it, but not render.*
 15. What's an ORM?
+  * Object Relational Mapping --ActiveRecord is an ORM. *
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+  * ActiveRecord *
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+  *
+    ```ruby
+      get '/restaraunts' do
+        @restaurants = Restaurant.all
+        erb :index
+      end
+      ^^ This renders the index page
+      
+      get '/restaurants/:id' do
+        @restaurant = Restaurant.find(params[:id])
+        erb :show
+      end
+      ^^ This renders an individual restaurant page.
+      
+      get '/restaurants/new' do
+        erb :new
+      end
+      ^^ This renders a page where a user can add a new restaurant.
+      
+      post '/restaurants' do
+        Restaurant.create(params[:restaraunt]
+        redirect '/restaurant'
+      end
+      ^^ This submits the new info from the form and posts it to /restaurants to be persisted.

```

Comment:
I really like what you're doing here. One minor comment on this route: I would say that the user submits the form which sends a POST request to `/restaurants`. This is the action that request hits that actually persists it to the database. Another way to say it would be that this action doesn't submit the info, it accepts it. Let me know if that distinction is unclear.